LambdaEngine/Include/ECS/ECSCore.h

```diff
@@ -112,9 +113,9 @@ namespace LambdaEngine
 		/*	Create component immediately, but hold off on registering and publishing it until the end of the frame.
 			This is to prevent concurrency issues. Publishing a component means pushing entity IDs to IDVectors,
 			and there is no guarentee that no one is simultaneously reading from these IDVectors. */
-		m_ComponentsToRegister.PushBack({entity, Comp::s_TID});
-
-		return m_ComponentStorage.AddComponent<Comp>(entity, component);
+		Comp& comp = m_ComponentStorage.AddComponent<Comp>(entity, component);
```

Comment:
Is there a reason you changed the order of adding the component and enqueuing its registration?

Reply:
I have not actually changed that. Maybe when I merged from master? Or the change was made on the master while I was working on this branch? Either way, let me know what is the correct code.

Reply:
Do it the 'old' way, register.PushBack first, then return storage.AddComp